File: phishing-detector/src/main/java/com/phishing/app/payload/DashboardData.java
package com.phishing.app.payload;

import java.io.Serializable;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class DashboardData implements Serializable {

    private CardData blackListedUrlsCard;
    private CardData usersCard;
    private CardData urlVerificationCard;
    private CardData phishingUrlsCard;
    private List<UserDetails> userDetailsList;
    private List<Site> siteList;
}

File: front-end/src/app/model/dashboard-data.ts
import { CardData } from "./card-data";
import { Site } from "./site";
import { UserDetails } from "./user-details";

export class DashboardData {
    blackListedUrlsCard: CardData;
    usersCard: CardData;
    urlVerificationCard: CardData;
    phishingUrlsCard: CardData;
    userDetailsList: Array<UserDetails>;
    siteList: any;
    userUrlRequestList: any;
}
